### ISSUE TITLE
StableDiffusionXLControlNetImg2ImgPipeline often fails to respect "pose" control images

### ISSUE THREAD

**Issue by Clement-Lelievre** [huggingface/diffusers#8930]
### Describe the bug

Using [StableDiffusionXLControlNetImg2ImgPipeline](https://huggingface.co/docs/diffusers/en/api/pipelines/controlnet_sdxl#diffusers.StableDiffusionXLControlNetImg2ImgPipeline), and passing a "pose" control image often fails to produce an output image that maintains the pose.
I couldn't find much info about this pipeline used with a pose image; I'd like to know whether the problem comes from the underlying pipe not being able to run an inference with this conditioning or if I'm doing something wrong, eg haven't found the right params.

Note that on the link above the example snippet uses a canny image; and that the `[controlnet model I'm using](https://huggingface.co/thibaud/controlnet-openpose-sdxl-1.0)` uses a pose image but the `StableDiffusionXLControlNetPipeline` pipeline instead of `StableDiffusionXLControlNetImg2ImgPipeline`.

### Reproduction

Using this image as reference image for the img2img part:
![image_quentin](https://github.com/user-attachments/assets/7ed0d32c-78ab-40d8-b448-03755bb6095d)

```python
import torch

from controlnet_aux import OpenposeDetector
from diffusers import ControlNetModel
from diffusers import StableDiffusionXLControlNetImg2ImgPipeline
from diffusers.utils import load_image
from PIL import Image

controlnet = ControlNetModel.from_pretrained(
    "thibaud/controlnet-openpose-sdxl-1.0",
    torch_dtype=torch.float16,
)
pipe = StableDiffusionXLControlNetImg2ImgPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    controlnet=controlnet,
    variant="fp16",
    use_safetensors=True,
    torch_dtype=torch.float16,
)
pipe.enable_model_cpu_offload()
openpose = OpenposeDetector.from_pretrained("lllyasviel/ControlNet")
pose_image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/person.png",
)
control_image = openpose(pose_image).resize((1024, 1024))
control_image.save('control.png')

prompt = "daiton style, daiton, A brave sheriff with a star badge, wearing a cowboy hat and spurs, standing confidently, illustration style, minimalist, illustration style, minimalist, solid color background"
negative_prompt = "blurry, ugly, messy weird"
image = Image.open(
    "/home/ubuntu/Developments/github.com/scenario-labs/phoenix/src/containers/inference_v2/image_quentin.png",
).resize((1024, 1024))
controlnet_conditioning_scale = 1.0

images = pipe(
    prompt=prompt,
    negative_prompt=negative_prompt,
    image=image,
    control_image=control_image,
    strength=1.0,
    num_inference_steps=30,
    controlnet_conditioning_scale=controlnet_conditioning_scale,
).images
images[0].save("from_diffusers.png")
```

Things I have tried:
- various params (eg guidance scale, more steps)
- other pose image
- thicker edges in the pose image
- other image sizes

### Logs

```shell
/home/ubuntu/anaconda3/envs/inference_v2/lib/python3.10/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
Loading pipeline components...: 100%|█| 7/7 [00:00<
100%|██████████████| 30/30 [00:14<00:00,  2.12it/s]
```

### System Info

- `diffusers` version: 0.27.2
- Platform: Linux-5.15.0-1048-aws-x86_64-with-glibc2.31
- Python version: 3.10.13
- PyTorch version (GPU?): 2.1.0 (True)
- Huggingface_hub version: 0.23.1
- Transformers version: 4.39.3
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: YES
- Using distributed or parallel set-up in script?: NO

### Who can help?

@yiyixuxu @sayakpaul @DN6

**Comment by Clement-Lelievre** [comment]
> So, IIUC, `StableDiffusionXLControlNetPipeline` with the same checkpoint (`thibaud/controlnet-openpose-sdxl-1.0`) does provide somewhat expected results for you.
> 
> Also, have you tried the Img2Img pipeline on some other poses just to see if similar observations hold?

yes and yes (similar observations hold, only a minority of outputs respect the pose)

**Comment by tolgacangoz** [comment]
If hand and face precisions are what you are looking for you can do this:
```diff
-control_image = openpose(pose_image).resize((1024, 1024))
+control_image = openpose(pose_image, hand_and_face=True).resize((1024, 1024))
```
These are the outputs with `controlnet_conditioning_scale=0.5` and `hand_and_face=True`, and `diffusers==0.29.1`:
![download](https://github.com/user-attachments/assets/9f680141-1790-4ed2-9e4d-3bdf4c3ae9bb)
![download (1)](https://github.com/user-attachments/assets/20b25ea4-e392-4d98-85c1-db0824ed08d0)

Most of the results seem reasonable to me :thinking:.

**Comment by Clement-Lelievre** [comment]
> Pinging @asomoza in case he has any insights here. Also, have you tried the same checkpoint but for text-to-image? If so, what are observations?
> 
> Meanwhile can you try https://huggingface.co/TencentARC/t2i-adapter-openpose-sdxl-1.0?

currently when loading https://huggingface.co/TencentARC/t2i-adapter-openpose-sdxl-1.0 as in the example snippet, I get `OSError: TencentARC/t2i-adapter-openpose-sdxl-1.0 does not appear to have a file named config.json.`

**Comment by asomoza** [comment]
I want to get the basic idea first, for what purpose are you using the initial image? if you use a strength of 1.0, you're basically telling the model to just ignore it.

So for example:

|1.0|0.8|0.5|
|---|---|---|
|![20240722114112_1638902058](https://github.com/user-attachments/assets/e1b52588-8b10-4cc0-b990-421ee66a6019)|![20240722114155_561233556](https://github.com/user-attachments/assets/bc13411b-2719-45d8-9d36-04507ecc6446)|![20240722114228_2021630740](https://github.com/user-attachments/assets/82637d97-a194-4e74-95d2-44d5c3d77721)|

So if you're going to use 1.0, you can just use a txt2img pipeline, which you wrote that it worked for you. 

Also the Pose controlnets for SDXL are worse than for SD 1.5, the only one that I find that works really good is the new one (Controlnet Union).

But using the same one that you have and with a strength of 1.0 which ignores the initial image, I get this result, but the model seems to be trained with low resolution poses, so you need to generate the pose control image with thicker lines:

|pose|generation|
|---|---|
|![20240722112208](https://github.com/user-attachments/assets/b433eaa6-b588-4335-b323-7f464a56ecb8)|![20240722114503_71587963](https://github.com/user-attachments/assets/fa2e2677-e804-42c3-bbd1-936f039bd445)|

but again, I want to understand your reason behind trying to do this with img2img, you're providing a detailed prompt and the initial image will interfere with the pose you're trying to do, it will only work if the pose is really similar to the initial image.

**Comment by Clement-Lelievre** [comment]
you're right, strength=1.0 was a bad example value. The actual value in my initial problem was strength=0.22 with num_inference_steps=30

What i would like to achieve is to get an output that:
- respects the text prompt
- gets the initial image character and or style
- gets the character from the previous item having the pose from control_image

**Comment by Clement-Lelievre** [comment]
I think so, are you able to get this kind of image from any pose (provided edges are thick enough) eg from the pose image I posted above:
![pose](https://github.com/user-attachments/assets/f7cd22be-5fa2-47b5-b7c8-582c623f113c)

**Comment by asomoza** [comment]
removed the face and hands because I don't think they're good for that character:

|pose|result|
|---|---|
|![20240722122217](https://github.com/user-attachments/assets/8cb7629c-a0eb-4b54-b25a-b86d350cd201)|![20240722122207](https://github.com/user-attachments/assets/14883a98-b57f-4e47-8c6d-10f6a30f7a9b)|

**Comment by Clement-Lelievre** [comment]
> removed the face and hands because I don't think they're good for that character:
> pose 	result
> ![20240722122217](https://github.com/user-attachments/assets/8cb7629c-a0eb-4b54-b25a-b86d350cd201) 	![20240722122207](https://github.com/user-attachments/assets/14883a98-b57f-4e47-8c6d-10f6a30f7a9b)

wow that looks good! Could you please share the snippet for this?
Thank you!

**Comment by asomoza** [comment]
sure, here it is:

```python
import torch

from diffusers import AutoencoderKL, ControlNetModel, DPMSolverMultistepScheduler, StableDiffusionXLControlNetPipeline
from diffusers.utils import load_image

controlnet = ControlNetModel.from_pretrained(
    "OzzyGT/controlnet-openpose-sdxl-1.0", variant="fp16", torch_dtype=torch.float16
)

pipe = StableDiffusionXLControlNetPipeline.from_pretrained(
    "path/repo of model",
    torch_dtype=torch.float16,
    variant="fp16",
    controlnet=controlnet,
).to("cuda")

pipe.load_ip_adapter(
    "h94/IP-Adapter",
    subfolder="sdxl_models",
    weight_name="ip-adapter-plus_sdxl_vit-h.safetensors",
    image_encoder_folder="models/image_encoder",
)
scale_config = [
    {
        "up": {"block_0": [0.8, 0.9, 0.0]},
    }
]
pipe.set_ip_adapter_scale(scale_config)

ipa_image = load_image("source.png")
pose = load_image("pose.png")


image = pipe(
    prompt="daiton style, daiton, A brave sheriff with a star badge, wearing a cowboy hat and spurs, illustration style, minimalist, illustration style, minimalist, solid color background",
    negative_prompt="",
    guidance_scale=6.5,
    num_inference_steps=25,
    image=pose,
    controlnet_conditioning_scale=1.0,
    control_guidance_end=0.9,
    ip_adapter_image=ipa_image,
).images[0]

image.save("/result.png")
```

Just in case, the controlnet its the same (thibaud) but in safetensors and saved with fp16. Also the model I used in my examples was `_CHEYENNE_` if you want to test it with that one. 

You'll need to replace the model repo/path and the path to the images.
